App didn't open on Android 10

**The bug**
App doesn't open after update to newest version, Android shows message 'App repeatedly gets closed'

**Clearing the storage fixed the issue**, but I'm filing anyway, because clearing the storage doesn't feel right. Maybe the following crash stacktrace can be of some use.

**To Reproduce**
Tried to open the app :)

```bash
[me@localhost ~]$ adb logcat guse.opentrack:E AndroidRuntime:E *:S
--------- beginning of main
--------- beginning of system
09-09 10:44:35.332 30513 30513 E guse.opentrack: Unknown bits set in runtime_flags: 0x20000
09-09 10:44:35.334 30513 30513 E guse.opentrack: Not starting debugger since process cannot load the jdwp agent.
--------- beginning of crash
09-09 10:44:35.373 30513 30513 E AndroidRuntime: FATAL EXCEPTION: main
09-09 10:44:35.373 30513 30513 E AndroidRuntime: Process: de.dennisguse.opentracks, PID: 30513
09-09 10:44:35.373 30513 30513 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider de.dennisguse.opentracks.content.provider.ShareContentProvider: java.lang.RuntimeException: Not supported
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread.installProvider(ActivityThread.java:7305)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread.installContentProviders(ActivityThread.java:6765)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6615)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread.access$1600(ActivityThread.java:232)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1953)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7697)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: Caused by: java.lang.RuntimeException: Not supported
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at de.dennisguse.opentracks.content.data.Track$Id.toString(Track.java:156)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at java.lang.String.valueOf(String.java:2924)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at de.dennisguse.opentracks.content.provider.CustomSQLiteOpenHelper.upgradeFrom25to26(CustomSQLiteOpenHelper.java:169)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at de.dennisguse.opentracks.content.provider.CustomSQLiteOpenHelper.onUpgrade(CustomSQLiteOpenHelper.java:71)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:417)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:317)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at de.dennisguse.opentracks.content.provider.CustomContentProvider.onCreate(CustomContentProvider.java:84)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at de.dennisguse.opentracks.content.provider.CustomContentProvider.onCreate(CustomContentProvider.java:71)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.content.ContentProvider.attachInfo(ContentProvider.java:2154)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.content.ContentProvider.attachInfo(ContentProvider.java:2128)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at de.dennisguse.opentracks.content.provider.ShareContentProvider.attachInfo(ShareContentProvider.java:123)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	at android.app.ActivityThread.installProvider(ActivityThread.java:7294)
09-09 10:44:35.373 30513 30513 E AndroidRuntime: 	... 10 more
```
**Technical information**

```bash
[me@localhost ~]$ adb devices -l
List of devices attached
b6f09ad2               device usb:1-2 product:OnePlus5T model:ONEPLUS_A5010 device:OnePlus5T transport_id:11

[me@localhost ~]$ adb shell getprop ro.build.version.sdk
29
[me@localhost ~]$ adb shell getprop ro.build.version.release
10
```
 - Device: [OnePlus 5T]
 - OS: [Android 10]
 - Build-Number: [ONEPLUS_A5010_43_200601]
 - Version [v3.9.5] (installed through F-Droid)

--- a/src/main/java/de/dennisguse/opentracks/SearchListActivity.java
+++ b/src/main/java/de/dennisguse/opentracks/SearchListActivity.java
@@ -415,7 +415,7 @@ public class SearchListActivity extends AbstractListActivity implements DeleteMa
         String icon = track.getIcon();
         String category = icon != null && !icon.equals("") ? null : track.getCategory();
 
-        resultMap.put(IS_RECORDING_FIELD, track.getId() == recordingTrackId);
+        resultMap.put(IS_RECORDING_FIELD, track.getId().equals(recordingTrackId));
         resultMap.put(IS_PAUSED_FIELD, recordingTrackPaused);
         resultMap.put(ICON_ID_FIELD, TrackIconUtils.getIconDrawable(icon));
         resultMap.put(ICON_CONTENT_DESCRIPTION_ID_FIELD, R.string.image_track);

--- a/src/main/java/de/dennisguse/opentracks/TrackEditActivity.java
+++ b/src/main/java/de/dennisguse/opentracks/TrackEditActivity.java
@@ -75,7 +75,7 @@ public class TrackEditActivity extends AbstractActivity implements ChooseActivit
         contentProviderUtils = new ContentProviderUtils(this);
         track = contentProviderUtils.getTrack(trackId);
         if (track == null) {
-            Log.e(TAG, "No track for " + trackId);
+            Log.e(TAG, "No track for " + trackId.getId());
             finish();
             return;
         }

--- a/src/main/java/de/dennisguse/opentracks/content/SearchEngine.java
+++ b/src/main/java/de/dennisguse/opentracks/content/SearchEngine.java
@@ -250,7 +250,7 @@ public class SearchEngine {
         score *= getTimeBoost(query, meanTimestamp);
 
         // Score the currently-selected track lower (user is already there, wouldn't be searching for it).
-        if (track.getId() == query.currentTrackId) {
+        if (track.getId().equals(query.currentTrackId)) {
             score *= CURRENT_TRACK_DEMOTION;
         }
 

--- a/src/main/java/de/dennisguse/opentracks/content/provider/CustomSQLiteOpenHelper.java
+++ b/src/main/java/de/dennisguse/opentracks/content/provider/CustomSQLiteOpenHelper.java
@@ -173,7 +173,7 @@ public class CustomSQLiteOpenHelper extends SQLiteOpenHelper {
                     Track.Id trackId = new Track.Id(cursor.getLong(trackIdIndex));
                     ContentValues cv = new ContentValues();
                     cv.put("uuid", UUIDUtils.toBytes(UUID.randomUUID()));
-                    db.update("tracks", cv, "_id = ?", new String[]{String.valueOf(trackId)});
+                    db.update("tracks", cv, "_id = ?", new String[]{String.valueOf(trackId.getId())});
                 } while (cursor.moveToNext());
             }
         }

--- a/src/main/java/de/dennisguse/opentracks/io/file/importer/AbstractFileTrackImporter.java
+++ b/src/main/java/de/dennisguse/opentracks/io/file/importer/AbstractFileTrackImporter.java
@@ -231,7 +231,7 @@ abstract class AbstractFileTrackImporter extends DefaultHandler implements Track
             trackId = new Track.Id(Long.parseLong(uri.getLastPathSegment()));
         } else {
             if (trackIds.size() > 0) {
-                throw new SAXException(createErrorMessage("Cannot import more than one track to an existing track " + importTrackId));
+                throw new SAXException(createErrorMessage("Cannot import more than one track to an existing track " + importTrackId.getId()));
             }
             trackId = importTrackId;
             contentProviderUtils.clearTrack(trackId);

--- a/src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java
+++ b/src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java
@@ -311,7 +311,7 @@ public class TrackRecordingService extends Service implements HandlerServer.Hand
     void resumeTrack(Track.Id trackId) {
         Track track = contentProviderUtils.getTrack(trackId);
         if (track == null) {
-            Log.e(TAG, "Ignore resumeTrack. Track " + trackId + " does not exists.");
+            Log.e(TAG, "Ignore resumeTrack. Track " + trackId.getId() + " does not exists.");
             return;
         }
 
@@ -335,7 +335,7 @@ public class TrackRecordingService extends Service implements HandlerServer.Hand
         Track track = contentProviderUtils.getTrack(recordingTrackId);
         if (track == null) {
             if (isRecording()) {
-                Log.w(TAG, "track is null, but recordingTrackId not -1L. " + recordingTrackId);
+                Log.w(TAG, "track is null, but recordingTrackId not -1L. " + recordingTrackId.getId());
                 updateRecordingState(null, true);
             }
             showNotification(false);

--- a/src/main/java/de/dennisguse/opentracks/util/IntentUtils.java
+++ b/src/main/java/de/dennisguse/opentracks/util/IntentUtils.java
@@ -87,7 +87,7 @@ public class IntentUtils {
         for (Track.Id trackId : trackIds) {
             Track track = contentProviderUtils.getTrack(trackId);
             if (track == null) {
-                Log.e(TAG, "TrackId " + trackId + " could not be resolved.");
+                Log.e(TAG, "TrackId " + trackId.getId() + " could not be resolved.");
                 continue;
             }
 

